# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/sync_interception.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/sync_interception.cc
--- a/security/sandbox/chromium/sandbox/win/src/sync_interception.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/sync_interception.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -14,7 +14,6 @@
 #include "sandbox/win/src/sandbox_nt_util.h"
 #include "sandbox/win/src/sharedmem_ipc_client.h"
 #include "sandbox/win/src/target_services.h"
-#include "mozilla/sandboxing/sandboxLogging.h"
 
 namespace sandbox {
 
@@ -47,8 +46,8 @@
     return SBOX_ERROR_GENERIC;
 
   SharedMemIPCClient ipc(ipc_memory);
-  ResultCode code = CrossCall(ipc, IPC_OPENEVENT_TAG, name, desired_access,
-                              answer);
+  ResultCode code =
+      CrossCall(ipc, IPC_OPENEVENT_TAG, name, desired_access, answer);
 
   return code;
 }
@@ -59,16 +58,12 @@
                                     POBJECT_ATTRIBUTES object_attributes,
                                     EVENT_TYPE event_type,
                                     BOOLEAN initial_state) {
-  NTSTATUS status = orig_CreateEvent(event_handle, desired_access,
-                                     object_attributes, event_type,
-                                     initial_state);
+  NTSTATUS status =
+      orig_CreateEvent(event_handle, desired_access, object_attributes,
+                       event_type, initial_state);
   if (status != STATUS_ACCESS_DENIED || !object_attributes)
     return status;
 
-  mozilla::sandboxing::LogBlocked("NtCreatEvent",
-                                  object_attributes->ObjectName->Buffer,
-                                  object_attributes->ObjectName->Length);
-
   // We don't trust that the IPC can work this early.
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return status;
@@ -78,25 +73,24 @@
       break;
 
     void* memory = GetGlobalIPCMemory();
-    if (memory == NULL)
+    if (!memory)
       break;
 
     OBJECT_ATTRIBUTES object_attribs_copy = *object_attributes;
     // The RootDirectory points to BaseNamedObjects. We can ignore it.
-    object_attribs_copy.RootDirectory = NULL;
+    object_attribs_copy.RootDirectory = nullptr;
 
-    wchar_t* name = NULL;
+    std::unique_ptr<wchar_t, NtAllocDeleter> name;
     uint32_t attributes = 0;
-    NTSTATUS ret = AllocAndCopyName(&object_attribs_copy, &name, &attributes,
-                                    NULL);
-    if (!NT_SUCCESS(ret) || name == NULL)
+    NTSTATUS ret =
+        AllocAndCopyName(&object_attribs_copy, &name, &attributes, nullptr);
+    if (!NT_SUCCESS(ret) || !name)
       break;
 
     CrossCallReturn answer = {0};
     answer.nt_status = status;
-    ResultCode code = ProxyCreateEvent(name, initial_state, event_type, memory,
-                                       &answer);
-    operator delete(name, NT_ALLOC);
+    ResultCode code = ProxyCreateEvent(name.get(), initial_state, event_type,
+                                       memory, &answer);
 
     if (code != SBOX_ALL_OK) {
       status = answer.nt_status;
@@ -105,12 +99,9 @@
     __try {
       *event_handle = answer.handle;
       status = STATUS_SUCCESS;
-    } __except(EXCEPTION_EXECUTE_HANDLER) {
+    } __except (EXCEPTION_EXECUTE_HANDLER) {
       break;
     }
-    mozilla::sandboxing::LogAllowed("NtCreateEvent",
-                                    object_attributes->ObjectName->Buffer,
-                                    object_attributes->ObjectName->Length);
   } while (false);
 
   return status;
@@ -120,15 +111,11 @@
                                   PHANDLE event_handle,
                                   ACCESS_MASK desired_access,
                                   POBJECT_ATTRIBUTES object_attributes) {
-  NTSTATUS status = orig_OpenEvent(event_handle, desired_access,
-                                   object_attributes);
+  NTSTATUS status =
+      orig_OpenEvent(event_handle, desired_access, object_attributes);
   if (status != STATUS_ACCESS_DENIED || !object_attributes)
     return status;
 
-  mozilla::sandboxing::LogBlocked("NtOpenEvent",
-                                  object_attributes->ObjectName->Buffer,
-                                  object_attributes->ObjectName->Length);
-  //
   // We don't trust that the IPC can work this early.
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return status;
@@ -138,24 +125,24 @@
       break;
 
     void* memory = GetGlobalIPCMemory();
-    if (memory == NULL)
+    if (!memory)
       break;
 
     OBJECT_ATTRIBUTES object_attribs_copy = *object_attributes;
     // The RootDirectory points to BaseNamedObjects. We can ignore it.
-    object_attribs_copy.RootDirectory = NULL;
+    object_attribs_copy.RootDirectory = nullptr;
 
-    wchar_t* name = NULL;
+    std::unique_ptr<wchar_t, NtAllocDeleter> name;
     uint32_t attributes = 0;
-    NTSTATUS ret = AllocAndCopyName(&object_attribs_copy, &name, &attributes,
-                                    NULL);
-    if (!NT_SUCCESS(ret) || name == NULL)
+    NTSTATUS ret =
+        AllocAndCopyName(&object_attribs_copy, &name, &attributes, nullptr);
+    if (!NT_SUCCESS(ret) || !name)
       break;
 
     CrossCallReturn answer = {0};
     answer.nt_status = status;
-    ResultCode code = ProxyOpenEvent(name, desired_access, memory, &answer);
-    operator delete(name, NT_ALLOC);
+    ResultCode code =
+        ProxyOpenEvent(name.get(), desired_access, memory, &answer);
 
     if (code != SBOX_ALL_OK) {
       status = answer.nt_status;
@@ -164,12 +151,9 @@
     __try {
       *event_handle = answer.handle;
       status = STATUS_SUCCESS;
-    } __except(EXCEPTION_EXECUTE_HANDLER) {
+    } __except (EXCEPTION_EXECUTE_HANDLER) {
       break;
     }
-    mozilla::sandboxing::LogAllowed("NtOpenEvent",
-                                    object_attributes->ObjectName->Buffer,
-                                    object_attributes->ObjectName->Length);
   } while (false);
 
   return status;